crates/core/database/src/models/messages/model.rs

```diff
@@ -302,19 +361,97 @@ impl Message {
                 .unwrap_or_default(),
             author: author_id,
             webhook: webhook.map(|w| w.into()),
-            flags: data.flags.map(|v| v as i32),
+            flags: data.flags,
             ..Default::default()
         };
 
         // Parse mentions in message.
         let mut mentions = HashSet::new();
-        if allow_mentions {
+        let mut role_mentions = HashSet::new();
+        let mut role_model_mentions = vec![];
+
+        if allow_mentions && server_id.is_some() {
             if let Some(content) = &data.content {
                 for capture in RE_MENTION.captures_iter(content) {
                     if let Some(mention) = capture.get(1) {
                         mentions.insert(mention.as_str().to_string());
                     }
                 }
+
+                // Second step of mass mention resolution.
+                for capture in RE_ROLE_MENTION.captures_iter(content) {
+                    let raw = capture.get(0).expect("No capture?").as_str();
+                    if raw == "@everyone" {
+                        mentions_everyone = true;
+                        // Can't break early here since we want to capture any role ids
+                    } else if raw == "@online" {
+                        mentions_online = true;
+                    } else if let Some(role_match) = capture.get(1) {
+                        role_mentions.insert(role_match.as_str().to_string());
+                    }
+                }
+
```

Comment:
This will capture escaped mentions, less importantly `\@everyone` (do we even care?) and more importantly:
```
`@everyone`, etc
```

Reply:
actually maybe we don't want it inside codeblocks, we could potentially do a find and replace of all codeblocks with an empty string and then run the mention find regex over that?